frontend/app/.server/web/validators/raoidc-session.validator.ts

```diff
@@ -0,0 +1,87 @@
+import { Session } from '@remix-run/node';
+
+import { inject, injectable } from 'inversify';
+
+import type { ServerConfig } from '~/.server/configs';
+import { TYPES } from '~/.server/constants';
+import type { LogFactory, Logger } from '~/.server/factories';
+import { RaoidcSessionInvalidException } from '~/.server/web/exceptions';
+import type { SessionService } from '~/.server/web/services';
+import { FetchFn, getFetchFn } from '~/utils/fetch-utils.server';
+import { IdToken, UserinfoToken, validateSession } from '~/utils/raoidc-utils.server';
+
+/**
+ * Provides functionality to validate CSRF tokens.
+ */
+export interface RaoidcSessionValidator {
+  /**
+   * Validates the CSRF token in the given request.
+   *
+   * @param request The request to validate the CSRF token from.
+   * @throws {Response} If the CSRF token is not found in the request or session, or if the tokens do not match.
+   */
+  validateRaoidcSession(request: Request): Promise<void>;
+}
```

Comment:
Nitpick: The documentation references CSRF but should reference the RAOIDC session instead.

Reply:
Not a nitpick.

Reply:
Fixed